### DISCUSSION TITLE
CSS and images do not load!!

### DISCUSSION THREAD

**Discussion by LonelyGuy12** [fastapi/fastapi#6415] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
from fastapi.templating import Jinja2Templates
from fastapi.responses import HTMLResponse
import uvicorn


templates = Jinja2Templates(directory="webpages")
app = FastAPI()

@app.route('/', methods=['GET'])
async def index(request: Request):
    return templates.TemplateResponse("coming_soon/index.html", {"request": request})


if __name__ == '__main__':
    uvicorn.run("app:app")
```

### Description

For some reason the CSS in the code doesn't load 😕

### Operating System

Windows

### Operating System Details

I am on Windows Insider Dev Build

### FastAPI Version

0.78.0

### Python Version

3.10.4

**Comment by JarroVGIT** [comment]
Not much to go on of course, but maybe you should serve your css file as a static file, so when referenced in your template it can be actually loaded? Otherwise, Fastapi (or actually, starlette in this case) won’t serve the CSS files on the path your browser would expect them.

**Comment by LonelyGuy12** [comment]
So.... what path will it be served on...?

**Comment by LonelyGuy12** [comment]
My directory structure
![image](https://user-images.githubusercontent.com/93421108/172072027-cee895aa-4712-45e5-af09-87394f9c7846.png)

**Comment by LonelyGuy12** [comment]
I am returning `./webpages/coming_soon/index.html`

**Comment by JarroVGIT** [comment, marked as answer]
This is all very well explained in the documentation,[ including an example ](https://fastapi.tiangolo.com/advanced/templates/#templates-and-static-files)of the issue you are raising here. 

Anyway, I created a MVP for you here: https://github.com/JarroVGIT/fastapi-github-issues/tree/master/4989

In short: you create an endpoint that serves a `StaticFiles` directory like so:

```python
app.mount("/coming_soon/static", StaticFiles(directory="webpages/coming_soon/static"), name="coming_soon_static")
```
If you run your Fastapi app, you can call static files from your `webpages/coming_soon/static` folder on endpoint `http://localhost:8000/coming_soon/static/file.ext` where file.ext can be any file. In this case, we have a style.css file in that folder.

Note: you currently have your `style.css` in the same folder as your `index.html`, and I would recommend to put it in a separate folder. Otherwise, your index.html would be directly accessible as plain text. That's why, I have put it in the `.webpages/coming_soon/static` folder.

In your `index.html` file, you can reference your stylesheet then like this:

```html
    <link href="{{ url_for('coming_soon_static', path='/style.css') }}" rel="stylesheet">
```

It references the `StaticFiles` named instance (`coming_soon_static`) and then asks for the file `/style.css`. This renders to:

```html
<link href="http://localhost:8000/coming_soon/static/style.css" rel="stylesheet">
```

The entire app looks like this:
```python
from fastapi import FastAPI, Request
from fastapi.responses import HTMLResponse
from fastapi.staticfiles import StaticFiles
from fastapi.templating import Jinja2Templates

app = FastAPI()

app.mount("/coming_soon/static", StaticFiles(directory="webpages/coming_soon/static"), name="coming_soon_static")


templates = Jinja2Templates(directory="webpages")


@app.get("/coming_soon", response_class=HTMLResponse)
async def coming_soon(request: Request):
    return templates.TemplateResponse("coming_soon/index.html", {"request": request})
```

This results in the following:

<img width="661" alt="image" src="https://user-images.githubusercontent.com/13659033/172114422-41ba07c5-21d7-4f66-a76f-33ff94c6503a.png">

**Comment by LonelyGuy12** [comment]
Thank You so much that worked!
